deepinv/physics/noise.py

```diff
@@ -114,7 +114,7 @@ def forward(self, x):
         :param torch.Tensor x: measurements
         :returns: noisy measurements
         """
-        y = torch.poisson(x / self.gain)
+        y = torch.poisson(torch.clip(x / self.gain, min=0.0))
```

Comment:
Not sure I agree with this: the noise should not impact the values of the measurement; what prevents one to add poisson noise to negative measurements?

Maybe with your usecase you could add this in the measurement operator?

Reply:
```python
>>> import torch
>>> torch.poisson(torch.tensor([-1.]))

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[5], [line 2](vscode-notebook-cell:?execution_count=5&line=2)
      [1](vscode-notebook-cell:?execution_count=5&line=1) import torch
----> [2](vscode-notebook-cell:?execution_count=5&line=2) torch.poisson(torch.tensor([-1.]))

RuntimeError: invalid Poisson rate, expected rate to be non-negative
```

The other option is to only apply noise to a mask of non-negative values?

Reply:
My bad I misunderstood the torch notation, the input should indeed be positive. Let's wait for @tachella's input :)

Reply:
I agree that poisson inputs should be positive but adding the clip might hide bugs - I think we should leave it to the user to pass positive rates

Reply:
How about a new parameter `PoissonNoise(clip_positive=False)`? Then users can easily choose this if they want to save them reimplementing PoissonNoise themselves. I see this as a common use-case when your NN might have negative hidden values (e.g. when using LeakyReLU) and the Poisson Noise applies the physics of clipping it and adding noise.

Reply:
ok, good idea, let's go with that optional parameter!